### PR TITLE
[HACCP-23] Added site map component.

### PR DESCRIPTION
**Pull request by spleshka**
Please provide a thorough description of your component and its usage with screenshots.

## Checklist before opening a PR

Before opening a PR, please make sure:

- [x] your work is in a branch based on and synced with the develop branch.
- [x] your scss and twig files are imported in the javascript file. Without this, Webpack won't be able to compile them.
- [x] your newComponent is imported in the index.js file.
- [x] your folder hierarchy follows the proper structure.
- [x] your code follows the coding standards and naming conventions defined.
- [x] your code has been tested as AA accessibility compliant to the WCAG 2.1 AA standard.
- [x] your code has been tested in all the browsers listed in the GDS browser's list, including IE11.
- [x] your code has been tested in all the different screensizes defined.
- [x] you have run the `yarn lint` command and fixed any linting errors.
- [x] you have updated the CHANGELOG.md file with your changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Upcoming release] - 1971-11-03
 
 ### Added
--
+- Site map component
```
Comment (c-ancia) [inline CHANGELOG.md:12]: Can you actually leave this empty as a template ready to be copy pasted and do a new "Upcoming release" section below it instead please? :)

File: src/components/general/SiteMap/siteMap.html.twig [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <p data-fsa-at="sitemap-link-{{ link.label|replace({" ": "-"})|lower }}">
+    {% include '@components/components/general/Link/link.html.twig' with {
+      link_url: link.href,
+      link_text: link.label,
+    } %}
+    </p>
+    {% if link.links %}
+      <div class="sitemap__second-level">
+      {% for sublink in link.links %}
+        <p data-fsa-at="sitemap-link-{{ sublink.label|replace({" ": "-"})|lower }}">
```
Comment (c-ancia) [inline src/components/general/SiteMap/siteMap.html.twig:13]: I think the data-fsa-at needs to be added on actionable elements. Do you think the event will bubble up to the p tag? 🤔

File: src/components/general/SiteMap/siteMap.html.twig [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{% if links %}
+<div class="sitemap" data-fsa-at="sitemap">
+  {% for link in links %}
+    <p data-fsa-at="sitemap-link-{{ link.label|replace({" ": "-"})|lower }}">
```
Comment (c-ancia) [inline src/components/general/SiteMap/siteMap.html.twig:4]: I think the data-fsa-at needs to be added on actionable elements. Do you think the event will bubble up to the p tag? 🤔

**Review by c-ancia — CHANGES_REQUESTED** [review]
Do you mind adding this component in one of the example pages please so it's visible also in a "use case" :)

File: src/components/general/SiteMap/siteMap.html.twig (hunk 2 above)
Comment (spleshka) [reply src/components/general/SiteMap/siteMap.html.twig:13]: Why do you think this is the case? I believe they just need a reliable selector to make sure it does not change across time.

File: CHANGELOG.md (hunk 1 above)
Comment (spleshka) [reply CHANGELOG.md:32]: Fixed.

File: src/components/general/SiteMap/siteMap.html.twig (hunk 3 above)
Comment (spleshka) [reply src/components/general/SiteMap/siteMap.html.twig:4]: Same response as before :)

**Review by c-ancia — APPROVED** [review]

**Pull request merged**